internal/services/apimanagement/api_management_workspace_resource.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package apimanagement
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2024-05-01/workspace"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ApiManagementWorkspaceModel struct {
+	WorkspaceId       string `tfschema:"workspace_id"`
+	ResourceGroupName string `tfschema:"resource_group_name"`
+	ServiceName       string `tfschema:"service_name"`
+	WorkspaceName     string `tfschema:"workspace_name"`
+}
+
+type ApiManagementWorkspaceResource struct{}
+
+var _ sdk.Resource = ApiManagementWorkspaceResource{}
+
+func (r ApiManagementWorkspaceResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"workspace_id": {
```

Comment:
This should be the name of the workspace
```suggestion
		"name": {
```